packages/form-field/src/FormFieldInputContainer/FormFieldInputContainer.tsx

```diff
@@ -68,12 +69,14 @@ export const FormFieldInputContainer = forwardRef<
 
             {contentEnd &&
               React.cloneElement(contentEnd, {
-                className: cx(
-                  iconClassName,
-                  getIconThemeStyles(theme),
-                  { [getIconDisabledThemeStyles(theme)]: disabled },
+                className: getContentEndStyles(
+                  theme,
+                  disabled,
```

Comment:
key addition is this

everything else is cleanup/refactoring/additional test coverage